src/scripts/synthetic_graph_generator.py

```diff
@@ -117,25 +62,105 @@ def _generate_single_grid_graph(size):
+            directed_tree.add_edges_from(tree.edges)
+            if nx.is_tree(directed_tree):
+                graph = directed_tree
+
+        case "complete":
+            graph = nx.complete_graph(no_of_vertices, create_using=nx.DiGraph())
+
+        case "watts-strogatz":
+            k = kwargs.get("k")
+            p = kwargs.get("p")
+
+            graph = nx.connected_watts_strogatz_graph(no_of_vertices, k, p, 1000)
+            filename = f"watts-strogatz_{no_of_vertices}_{len(graph.edges)}_{str(ratio[1]).replace(".", "")}_{str(p).replace(".", "")}"
+
+        case "grid":
+            graph = _generate_single_grid_graph(no_of_vertices)
+            while graph.out_degree(0) == 0:
+                graph = _generate_single_grid_graph(no_of_vertices)
+
+        case "random":
+            scalar = kwargs.get("scalar")
+            graph = nx.gnm_random_graph(no_of_vertices, scalar * no_of_vertices, directed=True)
+
+            while (not nx.is_weakly_connected(graph)) or (graph.out_degree(0) == 0):
+                graph = nx.gnm_random_graph(no_of_vertices, scalar * no_of_vertices, directed=True)
+            filename = f"random_{no_of_vertices}_{scalar * no_of_vertices}_{str(ratio[1]).replace(".", "")}"
+
+
+    if not filename:
+        filename = f"{graph_family}_{no_of_vertices}_{len(graph.edges)}_{str(ratio[1]).replace(".", "")}"
+
+    _save_graph_json(graph, no_of_vertices, ratio, filename)
+
+
+def generate_multiple_graphs(family: str, no_of_vertices, ratios):
     for num in no_of_vertices:
-        for ratio in ratios:
-            grid = _generate_single_grid_graph(num)
-            while grid.out_degree(0) == 0:
-                grid = _generate_single_grid_graph(num)
-            filename = f"grid_{num}x{num}_{len(grid.edges)}_{str(ratio[1]).replace(".", "")}"
-            _save_graph_json(grid, num, ratio, filename)
+        for r in ratios:
+            single_graph_generator(family, num, r)
+
+def generate_multiple_grids(no_of_vertices, ratios):
+    for num in no_of_vertices:
+        for r in ratios:
+            single_graph_generator("grid", num, r)
+
+def generate_multiple_random_graphs(no_of_vertices, ratios, scalars):
+    for num in no_of_vertices:
+        for r in ratios:
+            for s in scalars:
+                single_graph_generator("random", num, r, scalar=s)
+
+def generate_multiple_watts_strogatz_graphs(no_of_vertices, ratios, ks, ps):
+    for num in no_of_vertices:
+        for r in ratios:
+            for k in ks:
+                for p in ps:
+                    single_graph_generator("watts-strogatz", num, r, k=k, p=p)
 
 
 def main():
-    _generate_random_graphs()
-    _generate_families_of_graphs()
-    _generate_grids()
+    # PATHS, CYCLES, TREES, COMPLETE GRAPHS
+    families_of_graphs = ["path", "cycle", "random-tree", "complete"]
+    no_of_vertices = [10, 50, 100, 200, 500, 750, 1000]
+    ratios = [(0.9, 0.1), (0.8, 0.2), (0.66, 0.34), (0.5, 0.5), (0.2, 0.8)]
```

Comment:
Would like to see a full positive at least. Full negative would also be nice, since it Will on certsin types of graphs not have a negative cycle.